tests/idris2/reg036/Test.idr

```diff
@@ -0,0 +1,10 @@
+module Test
+
+interface Foo a where
+  bar : a -> {auto ok: ()} -> a
+
+foo : Void -> {auto ok: ()} -> Void
+foo = ?foo_hole
+
+baz : a -> b -> c -> {auto x : a} -> a
+baz {} = x
```

Comment:
I think I have found a bug involving shadowing:

```idris
foo : {a : Type} -> Void -> {a : a} -> Void -> {auto ok : ()} -> Void
foo _ {a = c} = ?foo_hole
-- ^   ^--- This should bind the second implicit `a`,
-- |        the one that comes after the first explicit argument
-- `-- there should be an implicitly bound `a : Type` here
```

gives me a hole with `c : Type` rather than `c : a`.

Reply:
I'm surprised to learn that this is legal Idris code. Does Idris have separate namespaces for term and type variables?

Reply:
Why shouldn't it be legal? It's not a namespace issue, it's classic
shadowing in a telescope: when checking the type `a` in the telescope
`pre -> (x : a) -> post`, the variables bound in `pre` are in scope
(but not `x` itself as no binding is self-referential) and when checking
`post`, the variables `pre, x` are in scope (thus any `x` in `pre` is
shadowed by the most-recently bound one).

Reply:
Ah, I see. Thanks for clarifying that for me!